node_scripts/flink/docker_main.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
```

Comment:
i feel like the entire node_scripts dir can be restructured a bit. A lot of the functions can be used for both spark and flink (like choose master, or wait for master, or setup ssh keys, etc...). The only main functions that need to be different are probably just __start_master__ and __start_worker__ and maybe some setup function to connect the cluster together right?

the directory is also structured in a confusing way. 

It is like this right now:
```
node_scripts
  /flink
    docker_main.sh
    /core
    /install
    ...
  docker_main.sh (for spark)
  /core (for spark)
  /install (for spark)
  ...
```

shouldn't it be more like this:
```
node_scripts
  /flink
  /spark
  /shared_stuff
  docker_main.sh
```

Reply:
My plan was to move all the stuff in the root of node_scripts now into a `spark/` directory like you have in your second example.

I'm a little wary of having a `node_scripts/shared_stuff` though. Flink and Spark have incredibly similar setup processes, but that is more or less just a coincidence (or more likely since they are both Apache projects and Flink borrowed from some of Sparks design).

My idea around `node_scripts` is that each software's sub-directory will contain all and only the files needed to completely set up that software on the nodes. That way in the future if Flink diverges in it's setup processes from Spark, we only need to change the Flink `node_scripts`.